documentation/api/query/v4/query.markdown

```diff
@@ -95,6 +95,19 @@ Most PuppetDB query endpoints support paged results via a set of shared URL para
 
 All queries return data with a content type of `application/json`. Each endpoint's page describes the format of its return data.
 
+### Rich data
+
+Puppet 6 supports
+[rich_data](https://github.com/puppetlabs/puppet-specifications/blob/master/language/types_values_variables.md#richdata)
+types like Timestamp and SemVer, and enables rich data by default.
+When rich data is enabled, readable string representations of rich
+data values may appear in the report resource event `old_value` and
+`new_value` fields, and in catalog parameter values.
+
+For example, a Timestamp value would be recorded in PuppetDB as a
+string like "2012-10-10T00:00:00.000000000 UTC", and a custom Sensor
+object would be recorded as a string like "Sensor({'temp' => '43'})".
```

Comment:
I would take out the Sensor example since the wire format for custom objects is not functional. You could instead use Binary or Sensitive/Deferred